core/src/main/java/hudson/search/Search.java

```diff
@@ -76,7 +76,7 @@ public void doIndex(StaplerRequest req, StaplerResponse rsp) throws IOException,
                     SuggestedItem target = find(index, query, smo);
                     if(target!=null) {
                         // found
-                        rsp.sendRedirect2(a.getUrl()+target.getUrl());
+                        rsp.sendRedirect2(req.getContextPath()+target.getUrl());
```

Comment:
> `goTo` requests an HTML page within Jenkins dashboard context and I need to do it from a different context

Sorry, not following. `goTo` simply simulates one HTTP request to a URL you select. What is the “context” you are talking about?